muspinsim/__main__.py

```diff
@@ -72,20 +116,30 @@ def main(use_mpi=False):
 
     is_fitting = mpi.broadcast(is_fitting)
 
+    if args.out_dir:
+        out_path = check_dir_path(args.out_dir)
+    else:
+        out_path = "./"
+
     if not is_fitting:
         # No fitting
         runner = ExperimentRunner(infile, {})
         runner.run()
 
         if mpi.is_root:
             # Output
-            runner.config.save_output()
+            runner.config.save_output(name=None, path=out_path)
     else:
         fitter = FittingRunner(infile)
-        fitter.run()
+        fitter.run(name=None, path=out_path)
 
         if mpi.is_root:
-            fitter.write_report()
+            if args.fitreport_path:
+                rep_path = check_dir_path(os.path.dirname(args.fitreport_path))
+                rep_fname = os.path.basename(args.fitreport_path)
```

Comment:
rep_fname needs some default too – running without this arg raises `UnboundLocalError: local variable 'rep_fname' referenced before assignment`